src/components/TodoItem.tsx

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { useState } from 'react';
+import cn from 'classnames';
+import { motion } from 'framer-motion';
+import { useDispatchContext } from '../context/GlobalContext';
+import EditTodoForm from './EditTodoForm';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+};
+
+const TodoItem = ({ todo }: Props) => {
```

Comment:
I'd prefer to extract all properties from the todo prop here OR - even better - to pass only required properties as props to the component.